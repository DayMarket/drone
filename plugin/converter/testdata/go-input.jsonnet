local goImage = if std.extVar("input.goImage") != "" then std.extVar("input.goImage") else "golang:1.18";
local pipeline_name = if std.extVar("input.pipelineName") != "" then std.extVar("input.pipelineName") else "development";

local pipeline(name) = {
  kind: 'pipeline',
  type: 'docker',
  name: name,
  withTrigger(trigger):: self + {
    trigger: trigger,
  },
  withVolumes(volumes):: self + {
    volumes: volumes,
  },
  withServices(services):: self + {
    services: services,
  },
  withSteps(steps):: self + {
    steps: steps,
  },
};

local step(name, image) = {
  name: name,
  image: image,
  withCommands(commands):: self + {
    commands: commands,
  },
  withWhen(when):: self + {
    when: when,
  },
  withVolumes(volumes):: self + {
    volumes: volumes,
  },
  withEnvs(envs):: self + {
    environment: envs,
  },
  withDeps(deps):: self + {
    depends_on: deps,
  },
  withSettings(settings):: self + {
    settings: settings,
  },
};

local golint = step(
  name='golint',
  image='golangci/golangci-lint:v1.52.0-alpine',
).withCommands(['golangci-lint run']);

local gotest = step(
  name='gotest',
  image=goImage,
).withCommands([
  'go test -v ./...',
]);

local volumes = [
  { name: 'docker', host: { path: '/var/run/docker.sock' } },
];

local development =
  pipeline(pipeline_name)
  .withTrigger({
    branch: ['dev'],
    event: ['pull_request', 'push'],
  })
  .withVolumes(volumes)
  .withSteps([
      golint,
      gotest,
  ]);

[
  development,
]