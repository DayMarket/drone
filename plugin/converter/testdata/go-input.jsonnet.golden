---
{
   "kind": "pipeline",
   "name": "template-1",
   "steps": [
      {
         "commands": [
            "golangci-lint run"
         ],
         "image": "golangci/golangci-lint:v1.52.0-alpine",
         "name": "golint"
      },
      {
         "commands": [
            "go test -v ./..."
         ],
         "image": "golang:1.19",
         "name": "gotest"
      }
   ],
   "trigger": {
      "branch": [
         "dev"
      ],
      "event": [
         "pull_request",
         "push"
      ]
   },
   "type": "docker",
   "volumes": [
      {
         "host": {
            "path": "/var/run/docker.sock"
         },
         "name": "docker"
      }
   ]
}
---
{
   "kind": "pipeline",
   "name": "template-2",
   "steps": [
      {
         "commands": [
            "golangci-lint run"
         ],
         "image": "golangci/golangci-lint:v1.52.0-alpine",
         "name": "golint"
      },
      {
         "commands": [
            "go test -v ./..."
         ],
         "image": "golang:1.19",
         "name": "gotest"
      }
   ],
   "trigger": {
      "branch": [
         "dev"
      ],
      "event": [
         "pull_request",
         "push"
      ]
   },
   "type": "docker",
   "volumes": [
      {
         "host": {
            "path": "/var/run/docker.sock"
         },
         "name": "docker"
      }
   ]
}
